src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -117,12 +117,18 @@ impl<'a> MacResult for ParserAnyMacro<'a> {
 
     fn make_stmts(self: Box<ParserAnyMacro<'a>>)
                  -> Option<SmallVector<ast::Stmt>> {
+        let parse_stmt = |parser: &mut Parser<'a>| -> ::parse::PResult<'a, _> {
```

Comment:
Good point, I'll replace the method `finish_parsing_statement` with something like `parse_full_stmt` and use that here instead of defining a closure here.